src/dbnode/storage/index/block.go

```diff
@@ -477,11 +478,12 @@ func (b *block) queryWithSpan(
 	}()
 
 	for iter.Next() {
-		if opts.LimitExceeded(size) {
+		if opts.SeriesLimitExceeded(size) || opts.DocsLimitExceeded(docsCount) {
 			break
 		}
 
 		batch = append(batch, iter.Current())
+		docsCount++
```

Comment:
Instead of incrementing ourselves (which doesn't take into account concurrent blocks querying and incrementing the docs count, which happens since blocks are queried in parallel) I think we actually want to return "docs" from "addQueryResults(...)" and the underlying "Results" struct which is tracking "TotalDocsCount" globally across the blocks that we query against.

Reply:
This is why "size" is returned from "addQueryResults" instead of tracking it ourselves.

Reply:
Yeah - I was trying to avoid having to propagate back the docs count in the same manner we do w/ size just for simplicity, but agreed we actually should just do that so it is consistent in how we track both of them back from the downstream `AddDocuments` call which internally does the locking